src/popup/views/messageEditor/components/messageSliderControls/component.tsx

```diff
@@ -0,0 +1,122 @@
+import { SettingsType } from "@src/lib/utilities/googleStorage";
+import { sendMessageToTab } from "@src/shared/utils";
+import React from "react";
+
+export interface SliderControlsProps {
+    setLiveChanges: React.Dispatch<React.SetStateAction<SettingsType>>;
+    liveChanges: SettingsType;
+    setIsEditing: React.Dispatch<React.SetStateAction<boolean>>;
+}
+export type colorSetting = "messageColor" | "textColor";
+
+export function MessageSliderControls({
+    setLiveChanges,
+    liveChanges,
+    setIsEditing,
+}: SliderControlsProps): JSX.Element {
+    // using this to create data we need so we can map through and have clean callback
+    class InputSetting {
+        name = "";
+        id: keyof SettingsType = "messageMaxWidthStyle";
+        valueType: "px" | "%" = "px";
+        bgColor = "";
+
+        constructor(settingsProperties: InputSetting) {
+            this.name = settingsProperties.name;
+            this.id = settingsProperties.id;
+            this.valueType = settingsProperties.valueType;
+            this.bgColor = settingsProperties.bgColor;
+        }
+    }
+    const inputSettings = [
+        new InputSetting({
+            name: "Message Width",
```

Comment:
we should make the text unselectable so users can't highlight things like `Message Width`